File: clients/Cpp/model/Camera.cpp
#include "Camera.hpp"

Camera::Camera() { }
Camera::Camera(Vec2Float center, float rotation, float attack, float distance, bool perspective) : center(center), rotation(rotation), attack(attack), distance(distance), perspective(perspective) { }
Camera Camera::readFrom(InputStream& stream) {
    Camera result;
    result.center = Vec2Float::readFrom(stream);
    result.rotation = stream.readFloat();
    result.attack = stream.readFloat();
    result.distance = stream.readFloat();
    result.perspective = stream.readBool();
    return result;
}
void Camera::writeTo(OutputStream& stream) const {
    center.writeTo(stream);
    stream.write(rotation);
    stream.write(attack);
    stream.write(distance);
    stream.write(perspective);
}

File: clients/Scala/src/main/scala/model/Camera.scala
package model

import util.StreamUtil

case class Camera(center: model.Vec2Float, rotation: Float, attack: Float, distance: Float, perspective: Boolean) {
    def writeTo(stream: java.io.OutputStream) {
        center.writeTo(stream)
        StreamUtil.writeFloat(stream, rotation)
        StreamUtil.writeFloat(stream, attack)
        StreamUtil.writeFloat(stream, distance)
        StreamUtil.writeBoolean(stream, perspective)
    }
}
object Camera {
    def readFrom(stream: java.io.InputStream): Camera = Camera(
        model.Vec2Float.readFrom(stream)
        ,
        StreamUtil.readFloat(stream)
        ,
        StreamUtil.readFloat(stream)
        ,
        StreamUtil.readFloat(stream)
        ,
        StreamUtil.readBoolean(stream)
        )
}
